iris-mpc-cpu/src/hnsw/searcher.rs

```diff
@@ -422,6 +484,53 @@ impl HnswSearcher {
         (links, set_ep)
     }
 
+    /// Two-step variant of `insert_from_search_results`: prepare.
```

Comment:
nit: Would it make sense to leave a to-do to refactor `insert_from_search_results` using these new functions?

Reply:
Done!